import_obfuscation.go

```diff
@@ -0,0 +1,473 @@
+	"go.cuinfo.",
+	"go.info.",
+	// string names be what they be
+	"go.string.",
+
+	// skip entrypoint symbols
+	"main.init.",
+	"main..stmp",
+}
+
+var entrypointSyms = [...]string{
+	"main.main",
+	"main..inittask",
+}
+
+var skipSubstrs = [...]string{
+	// skip cgo symbols
+	"_cgo_",
+	"_Cfunc_",
+	"_Ctype_",
+	"_Cgo_ptr",
+	// skip test symbols
+	"_test.",
+}
+
+var symPrefixes = [...]string{
+	"go.builtin.",
+	"go.itab.",
+	"go.itablink.",
+	"go.interface.",
+	"go.map.",
+	"gofile..",
+	"type..eq.",
+	"type..eqfunc.",
+	"type..hash.",
+	"type..importpath.",
+	"type..namedata.",
+	"type.",
+}
+
+// splitSymbolPrefix returns the symbol name prefix Go uses
+// to help designate the type of the symbol, and the rest of
+// the symbol name. Additionally, a bool is returned that
+// signifies whether garbling the symbol name should be skipped.
+func splitSymbolPrefix(symName string) (string, string, bool) {
+	if symName == "" {
+		return "", "", true
+	}
+
+	for _, prefix := range skipPrefixes {
+		if strings.HasPrefix(symName, prefix) {
+			return "", "", true
+		}
+	}
+
+	for _, entrySym := range entrypointSyms {
+		if symName == entrySym {
+			return "", "", true
+		}
+	}
+
+	for _, substr := range skipSubstrs {
+		if strings.Contains(symName, substr) {
+			return "", "", true
+		}
+	}
+
+	for _, prefix := range symPrefixes {
+		if strings.HasPrefix(symName, prefix) {
+			return symName[:len(prefix)], symName[len(prefix):], false
+		}
+	}
+
+	return "", symName, false
+}
+
+func privateImportIndex(symName string, privateImports []string) (int, int) {
+	firstOff, l := -1, 0
+	for _, privateImport := range privateImports {
+		// search for the package name plus a period if the
+		// package name doesn't have slashes, to minimize the
+		// likelihood that the package isn't matched as a
+		// substring of another ident name.
+		// ex: privateImport = main, symName = "domainname"
+		var noSlashes bool
+		if !strings.ContainsRune(privateImport, '/') {
+			privateImport += "."
+			noSlashes = true
+		}
+
+		off := strings.Index(symName, privateImport)
+		if off == -1 {
+			continue
+			// check that we didn't match inside an import path. If the
+			// byte before the start of the match is not a small set of
+			// symbols that can make up a symbol name, we must have matched
+			// inside of an ident name as a substring. Or, if the byte
+			// before the start of the match is a forward slash, we are
+			// definitely inside of an input path.
+		} else if off != 0 && (!isSymbol(symName[off-1]) || symName[off-1] == '/') {
+			continue
+		}
+
+		if off < firstOff || firstOff == -1 {
+			firstOff = off
+			l = len(privateImport)
+			if noSlashes {
+				l--
+			}
+		}
+	}
+
+	if firstOff == -1 {
+		return -1, 0
+	}
+
+	return firstOff, l
+}
+
+func isSymbol(c byte) bool {
+	return c == 32 || // ' '
+		(c >= 40 && c <= 42) || c == 44 || // '(', ')', '*', ','
+		c == 91 || c == 93 || c == 95 || // '[', ']', '_'
+		c == 123 || c == 125 // '{', '}'
```

Comment:
I don't get this; this could literally just be:

```
switch c {
case ' ', '(', ')', '*', ...:
    return true
}
return false
```

Reply:
*facepalm* right, forgot you could do that tbh